DataFormats/Detectors/TRD/include/DataFormatsTRD/TrapConfig3.h

```diff
@@ -0,0 +1,619 @@
+  kTPQS1,
+  kTPQE1,
+  kEBD,
+  kEBAQA,
+  kEBSIA,
+  kEBSF,
+  kEBSIM,
+  kEBPP,
+  kEBPC,
+  kFPTC,
+  kFPNP,
+  kFPCL,
+  kFGTA,
+  kFGTB,
+  kFGCL,
+  kFTAL,
+  kFTLL,
+  kFTLS,
+  kFCW1,
+  kFCW2,
+  kFCW3,
+  kFCW4,
+  kFCW5,
+  kTPFP,
+  kTPHT,
+  kADCMSK,
+  kADCINB,
+  kADCDAC,
+  kADCPAR,
+  kADCTST,
+  kSADCAZ,
+  kPASADEL,
+  kPASAPHA,
+  kPASAPRA,
+  kPASADAC,
+  kPASASTL,
+  kPASAPR1,
+  kPASAPR0,
+  kSADCTRG,
+  kSADCRUN,
+  kSADCPWR,
+  kL0TSIM,
+  kSADCEC,
+  kSADCMC,
+  kSADCOC,
+  kSADCGTB,
+  kSEBDEN,
+  kSEBDOU,
+  kSML0,
+  kSML1,
+  kSML2,
+  kSMMODE,
+  kNITM0,
+  kNITM1,
+  kNITM2,
+  kNIP4D,
+  kARBTIM,
+  kIA0IRQ0,
+  kIA0IRQ1,
+  kIA0IRQ2,
+  kIA0IRQ3,
+  kIA0IRQ4,
+  kIA0IRQ5,
+  kIA0IRQ6,
+  kIA0IRQ7,
+  kIA0IRQ8,
+  kIA0IRQ9,
+  kIA0IRQA,
+  kIA0IRQB,
+  kIA0IRQC,
+  kIRQSW0,
+  kIRQHW0,
+  kIRQHL0,
+  kIA1IRQ0,
+  kIA1IRQ1,
+  kIA1IRQ2,
+  kIA1IRQ3,
+  kIA1IRQ4,
+  kIA1IRQ5,
+  kIA1IRQ6,
+  kIA1IRQ7,
+  kIA1IRQ8,
+  kIA1IRQ9,
+  kIA1IRQA,
+  kIA1IRQB,
+  kIA1IRQC,
+  kIRQSW1,
+  kIRQHW1,
+  kIRQHL1,
+  kIA2IRQ0,
+  kIA2IRQ1,
+  kIA2IRQ2,
+  kIA2IRQ3,
+  kIA2IRQ4,
+  kIA2IRQ5,
+  kIA2IRQ6,
+  kIA2IRQ7,
+  kIA2IRQ8,
+  kIA2IRQ9,
+  kIA2IRQA,
+  kIA2IRQB,
+  kIA2IRQC,
+  kIRQSW2,
+  kIRQHW2,
+  kIRQHL2,
+  kIA3IRQ0,
+  kIA3IRQ1,
+  kIA3IRQ2,
+  kIA3IRQ3,
+  kIA3IRQ4,
+  kIA3IRQ5,
+  kIA3IRQ6,
+  kIA3IRQ7,
+  kIA3IRQ8,
+  kIA3IRQ9,
+  kIA3IRQA,
+  kIA3IRQB,
+  kIA3IRQC,
+  kIRQSW3,
+  kIRQHW3,
+  kIRQHL3,
+  kCTGDINI,
+  kCTGCTRL,
+  kMEMRW,
+  kMEMCOR,
+  kDMDELA,
+  kDMDELS,
+  kNMOD,
+  kNDLY,
+  kNED,
+  kNTRO,
+  kNRRO,
+  kNBND,
+  kNP0,
+  kNP1,
+  kNP2,
+  kNP3,
+  kC08CPU0,
+  kC09CPU0, // will be q2 pid window settings.
+  kC10CPU0,
+  kC11CPU0,
+  kC12CPUA,
+  kC13CPUA,
+  kC14CPUA,
+  kC15CPUA,
+  kC08CPU1,
+  kC09CPU1, // version of trap code.
+  kC10CPU1,
+  kC11CPU1,
+  kC08CPU2,
+  kC09CPU2,
+  kC10CPU2,
+  kC11CPU2,
+  kC08CPU3,
+  kC09CPU3,
+  kC10CPU3,
+  kC11CPU3,
+  kNES,
+  kNTP,
+  kNCUT,
+  kPASACHM,
+  kLastReg
+};
+
+class TrapRegInfo
+{
+  // class to store the parameters associated with a register
+  // some info related to the hardware, some to the packing/unpacking we do here.
+ public:
+  TrapRegInfo();
+  TrapRegInfo(const std::string& name, int addr, int nBits, int base, int wordoffset, bool ignorechange, uint32_t max);
+
+  ~TrapRegInfo();
+
+  void init(const std::string& name, int addr, int nBits, int base, int wordnumber, bool ignorechange, uint32_t max);
+
+  // getters and setters, this is just a storage class.
+  std::string getName() { return mName; }
+  unsigned short getAddr() { return mAddr; }
+  unsigned short getNbits() { return mNbits; }
+  unsigned int getBase() { return mBase; }
+  unsigned int getWordNumber() { return mWordNumber; }
+  unsigned int getDataWordNumber() { return mDataWordNumber; }
+  unsigned int getShift() { return mShift; }
+  uint32_t getMask() { return mMask; }
+  uint32_t getMax() { return mMax; }
+  bool getIgnoreChange() { return mIgnoreChange; }
```

Comment:
Generally, it would feel right to mark all getters as const.
Also in TrapConfig.